reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportInstanceIT.kt

```diff
@@ -0,0 +1,290 @@
+        Assert.assertEquals(
+            "Success",
+            updatedReportInstanceStatus
+        )
+        Assert.assertEquals(
+            "Operation completed",
+            updatedReportInstanceText
+        )
+    }
+
+    fun `test update report instance status with invalid instance id`() {
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val updateStatus = """
+            {
+                "status":"Success",
+                "statusText":"Operation completed"
+            }
+        """.trimIndent()
+
+        val dummyInstanceId = "abcdefghijk123"
+        val reportDefinitionUpdateResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/instance/$dummyInstanceId",
+            updateStatus,
+            RestStatus.NOT_FOUND.status
+        )
+        validateErrorResponse(reportDefinitionUpdateResponse, RestStatus.NOT_FOUND.status)
+    }
+
+    fun `test update report instance status with invalid update status`() {
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val onDemandRequest = """
+            {}
+        """.trimIndent()
+        val onDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$reportDefinitionId",
+            onDemandRequest,
+            200
+        )
+
+        val updateStatus = """
+            {
+                "invalidStatus":"Success",
+                "statusText":"Operation completed"
+            }
+        """.trimIndent()
+        val reportInstance = onDemandResponse.get("reportInstance").asJsonObject
+        val reportInstanceId = reportInstance.get("id").asString
+
+        val reportDefinitionUpdateResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/instance/$reportInstanceId",
+            updateStatus,
+            RestStatus.BAD_REQUEST.status
+        )
+        val error = reportDefinitionUpdateResponse.get("error").asJsonObject
```

Comment:
may be use testhelper function for validating error response?

Reply:
`validateErrorResponse` checks for `status_exception` and not `illegal_argument_exception`, I can wait for Josh's PR to merge but I don't think we should touch the same helper function in 2 separate PRs